pre_processing/read_aatsr_dimensions.F90

```diff
@@ -80,7 +83,7 @@ end subroutine get_aatsr_dimension
    logical,                        intent(in)  :: verbose
 
    character(len=path_length,kind=c_char) :: l1b_file_C
-   integer(c_short)                       :: tmp_dynght, half_orbit
+   integer(c_short)                       :: tmp_dynght, half_orbit, c_dynght
```

Comment:
I don't think you needed to make an extra variable here, as tmp_dynght is only passed to C routines or considered in if statements.